### PR TITLE
Remove Indirections and EMF Profiles from the metamodel

### PR DESCRIPTION
**Pull request by Entenwilli**
This PR removes all dependencies on indirections and EMF Profiles.
Due to this the following projects have been removed:
- `org.dataflowanalysis.pcm.extension.editor.sirius`: From what I could glean, this project was responsible for the application of EMF Profiles from within the model editor
- `org.dataflowanalysis.pcm.extension.model.profile`: The project containing the EMF profile definition

Furthermore, the following projects have been changed (in Addition to removing the dependencies to the indirections and modellingversioning projects):
- `org.dataflowanalysis.pcm.extension.model`: The dataflow confidentiality metamodel contained a smaller reference to additional behavior added by indirections to connections between resource environments
- `org.dataflowanalysis.pcm.extension.queryutils`: Removed utility functions that related to the removed model components

### CODE REVIEW

**Comment by Entenwilli** [comment]
@Nicolas-Boltz This PR is also ready. Please also check, if the model behaves as it should. In my testing, I've found no issues

File: bundles/org.dataflowanalysis.pcm.extension.dictionary.characterized/model/DataDictionaryCharacterized.genmodel [hunk 1]
```diff
@@ -19,8 +19,8 @@
   <modelPluginVariables>CDO=org.eclipse.emf.cdo</modelPluginVariables>
   <editPluginVariables>CDO_EDIT=org.eclipse.emf.cdo.edit</editPluginVariables>
   <genPackages prefix="DataDictionaryCharacterized" basePackage="org.dataflowanalysis.pcm.extension.dictionary.characterized"
-      disposableProviderFactory="true" extensibleProviderFactory="true" childCreationExtenders="true"
```
Comment (Nicolas-Boltz) [inline bundles/org.dataflowanalysis.pcm.extension.dictionary.characterized/model/DataDictionaryCharacterized.genmodel:22]: These changes seem to be unrelated and might be an automatic change from opening the genmodel.
Is there a reason for this change?

File: bundles/org.dataflowanalysis.pcm.extension.dictionary/model/DataDictionary.genmodel [hunk 2]
```diff
@@ -17,8 +17,8 @@
   <modelPluginVariables>CDO=org.eclipse.emf.cdo</modelPluginVariables>
   <editPluginVariables>CDO_EDIT=org.eclipse.emf.cdo.edit</editPluginVariables>
   <genPackages prefix="DataDictionary" basePackage="org.dataflowanalysis.pcm.extension.dictionary"
-      disposableProviderFactory="true" extensibleProviderFactory="true" childCreationExtenders="true"
```
Comment (Nicolas-Boltz) [inline bundles/org.dataflowanalysis.pcm.extension.dictionary/model/DataDictionary.genmodel:20]: Same as above

File: bundles/org.dataflowanalysis.pcm.extension.dictionary.characterized/model/DataDictionaryCharacterized.genmodel (hunk 1 above)
Comment (Entenwilli) [reply bundles/org.dataflowanalysis.pcm.extension.dictionary.characterized/model/DataDictionaryCharacterized.genmodel:22]: GitHub Diff is a bit misleading here, I only set `generateExampleClass` to false, as it creates an empty test model project, which fails to build/test

File: bundles/org.dataflowanalysis.pcm.extension.dictionary/model/DataDictionary.genmodel (hunk 2 above)
Comment (Entenwilli) [reply bundles/org.dataflowanalysis.pcm.extension.dictionary/model/DataDictionary.genmodel:20]: See comment from above

**Comment by Nicolas-Boltz** [comment]
You tested all sirius and tree editors? Everything there works fine? 
Have you tried opening one/some of the testmodels with the (sirius)editors modifying something minor, saving and running the analysis? If not please do, to ensure that there are no hidden references in there/that the editors work properly.

If all of the above works we can merge.

**Review by Nicolas-Boltz — APPROVED** [review]
Looks good to me
